lib/Dialect/Comb/InferIntRangeInterfaceImpls.cpp

```diff
@@ -0,0 +1,309 @@
+//===- InferIntRangeInterfaceImpls.cpp - Integer range impls for comb -----===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// Implementation of the interval range analysis interface.
+// The overflow flags are not set for the comb operations since they is
+// no meaningful concept of overflow detection in comb.
+//
+//===----------------------------------------------------------------------===//
+
+#include "circt/Dialect/Comb/CombOps.h"
+
+#include "mlir/Interfaces/InferIntRangeInterface.h"
+#include "mlir/Interfaces/Utils/InferIntRangeCommon.h"
+
+using namespace mlir;
+using namespace mlir::intrange;
+using namespace circt;
+using namespace circt::comb;
+//===----------------------------------------------------------------------===//
+// AddOp
+//===----------------------------------------------------------------------===//
+
+void comb::AddOp::inferResultRanges(ArrayRef<ConstantIntRanges> argRanges,
+                                    SetIntRangeFn setResultRange) {
+  auto resultRange = argRanges[0];
+  for (size_t i = 1; i < argRanges.size(); ++i)
+    resultRange =
+        inferAdd({resultRange, argRanges[i]}, intrange::OverflowFlags::None);
```

Comment:
There's a `drop_front` method that you can use for these kind of things. That's super nit picky, but just so you know.
```suggestion
  for (auto arg : argRanges.drop_front())
    resultRange =
        inferAdd({resultRange, arg}, intrange::OverflowFlags::None);
```

Reply:
Integrated the change will close.